src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -570,6 +570,30 @@ async def build_pex(
     request: PexRequest, python_setup: PythonSetup, pex_subsystem: PexSubsystem
 ) -> BuildPexResult:
     """Returns a PEX with the given settings."""
+
+    if not request.interpreter_constraints:
+        # Blank ICs in the request means that the caller wants us to use the ICs configured
+        # for the resolve (or the global ICs, if using the builtin lockfile).
+        if isinstance(request.requirements, PexRequirements) and isinstance(
+            request.requirements.from_superset, Resolve
+        ):
+            request = dataclasses.replace(
+                request,
+                interpreter_constraints=InterpreterConstraints(
+                    python_setup.resolves_to_interpreter_constraints.get(
+                        request.requirements.from_superset.name,
+                        python_setup.interpreter_constraints,
+                    )
+                ),
+            )
+        elif isinstance(request.requirements, EntireLockfile):
+            request = dataclasses.replace(
+                request,
+                interpreter_constraints=InterpreterConstraints(
+                    python_setup.interpreter_constraints
+                ),
```

Comment:
I'm trying to understand why EntireLockfile only uses the default interpreter constraints instead of a resolve's constraints.

Is `EntireLockfile` only used for the default tool lockfiles?

Or could there be times where `EntireLockfile` represents a user resolve's lockfile, and therefore needs to use that resolve's interpreter_constraints?

Reply:
Maybe this block needs to be like this? Are there cases where this does not make sense?
```suggestion
        elif isinstance(request.requirements, EntireLockfile):
            request = dataclasses.replace(
                request,
                interpreter_constraints=InterpreterConstraints(
                    python_setup.resolves_to_interpreter_constraints.get(
                        request.requirements.lockfile.resolve_name,
                        python_setup.interpreter_constraints,
                    )
                ),
```

Reply:
Hmm EntireLockfile might be used if `run_against_entire_lockfile = True`, it's hard to follow in the code (but makes sense).  So I think you're right. Although in that case the ICs will be set from targets, so it might be moot.

Reply:
Done